platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c

```diff
@@ -251,28 +248,31 @@ static inline hrt_abstime hrt_calc_base_time(uint32_t newloadval)
 		base_time += loadval;
 	}
 
+	/* calculate the current time. if loadval < count this wraps around correctly */
+	curr_time = base_time + (loadval - count);
+
 	/* set new last counter val if needed */
 	if (newloadval != 0) {
```

Comment:
Yes, it would probably be cleaner. Then these would need to move out of the function:
	static volatile uint32_t loadval;
	static volatile hrt_abstime base_time;
	static volatile uint32_t last_count;

since they are set in both "counter reload" and "query" paths.

Perhaps next iteration some day in the future and not put in this same PR?